src/queue.c

```diff
@@ -5673,7 +5736,11 @@ _dispatch_queue_set_mainq_drain_state(bool arg)
 }
 
 void
+#if TARGET_OS_MAC
 _dispatch_main_queue_callback_4CF(mach_msg_header_t *msg DISPATCH_UNUSED)
+#else
+_dispatch_main_queue_callback_4CF(void)
```

Comment:
this seems like an unnecessary difference if we made the argument void* and just pass in NULL (given that it is not used)